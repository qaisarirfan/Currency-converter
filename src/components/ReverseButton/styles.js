import {StyleSheet} from "react-native"

// ReverseButton: Styling here
const styles = StyleSheet.create({
  button: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    marginVertical: 20,
  },
  buttonIcon: {
    marginRight: 10,
    width: 20,
    height: 20,
  },
  buttonText: {
    fontSize: 16,
    color: "#fff",
  },
})

export default styles
