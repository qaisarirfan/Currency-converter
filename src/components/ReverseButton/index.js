import React from "react"
import {Text, TouchableOpacity, Image} from "react-native"
import PropTypes from "prop-types"
import styles from "./styles"
import reverse from "../../assets/images/reverse.png"

// ReverseButton Component content
export const ReverseButton = ({onPress, text}) => (
  <TouchableOpacity
    testID="reverse_button"
    onPress={onPress}
    style={styles.button}>
    <Image source={reverse} style={styles.buttonIcon} resizeMode="contain" />
    <Text style={styles.buttonText}>{text}</Text>
  </TouchableOpacity>
)

// ReverseButton Proptypes
ReverseButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
}

// ReverseButton Default props
ReverseButton.defaultProps = {}

export default ReverseButton
