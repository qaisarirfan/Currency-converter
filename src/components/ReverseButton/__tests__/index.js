import React from "react"
import renderer from "react-test-renderer"
import ReverseButton from ".."


describe("<ReverseButton />", () => {
  const createProps = () => ({
    onPress: () => {},
    text: "test"
  })
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<ReverseButton {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
