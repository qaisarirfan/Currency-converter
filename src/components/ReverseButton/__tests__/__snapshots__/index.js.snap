// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReverseButton /> Does not explode 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 20,
      "opacity": 1,
    }
  }
  testID="reverse_button"
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 20,
        "marginRight": 10,
        "width": 20,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;
